apps/framework-cli/src/infrastructure/olap/clickhouse/queries.rs

```diff
@@ -129,30 +173,378 @@ impl<'a> TryFrom<&'a str> for ClickhouseEngine {
+            } else if parts.len() >= 4 && !parts[1].is_empty() && !parts[2].is_empty() {
+                // Check if parts[1] and parts[2] look like credentials (not format names)
+                // Common formats are: CSV, TSV, JSON, Parquet, etc. - all uppercase or mixed case
+                // If parts[3] looks like a format name and parts[1] doesn't, assume we have credentials
+                let possible_format = &parts[3].to_uppercase();
+                if possible_format == "CSV"
+                    || possible_format == "TSV"
+                    || possible_format == "JSON"
+                    || possible_format == "PARQUET"
+                    || possible_format == "AVRO"
+                    || possible_format == "ORC"
+                    || possible_format == "ARROW"
+                    || possible_format == "NATIVE"
+                    || possible_format == "JSONCOMPACT"
+                    || possible_format == "JSONEACHROW"
+                {
+                    // parts[1] and parts[2] are likely credentials
+                    // Note: parts[2] might be "[HIDDEN]" when parsed from SHOW CREATE TABLE
+                    (
+                        parts[3].clone(),
+                        Some(parts[1].clone()),
+                        Some(parts[2].clone()),
+                        4,
+                    )
+                } else {
+                    // No credentials, parts[1] is the format
+                    (parts[1].clone(), None, None, 2)
+                }
+            } else {
+                // No credentials, parts[1] is the format
+                (parts[1].clone(), None, None, 2)
+            };
+
+        // Parse optional compression (next parameter after format)
+        let compression = if parts.len() > extra_params_start && parts[extra_params_start] != "null"
+        {
+            Some(parts[extra_params_start].clone())
+        } else {
+            None
+        };
+
+        // Parse optional headers JSON (parameter after compression)
+        let headers =
+            if parts.len() > extra_params_start + 1 && parts[extra_params_start + 1] != "null" {
+                // Unescape the JSON string (reverse the escaping we did during serialization)
+                let unescaped = parts[extra_params_start + 1].replace("\\'", "'");
+                serde_json::from_str::<std::collections::HashMap<String, String>>(&unescaped).ok()
+            } else {
+                None
+            };
+
+        Ok(ClickhouseEngine::S3Queue {
+            s3_path,
+            format,
+            compression,
+            headers,
+            aws_access_key_id,
+            aws_secret_access_key,
+        })
+    }
+
+    /// Calculate a hash of non-alterable parameters for change detection
+    /// This allows us to detect changes in constructor parameters without storing sensitive data
+    pub fn non_alterable_params_hash(&self) -> String {
+        let mut hasher = DefaultHasher::new();
```

Comment:
I skimmed the docs on DefaultHasher https://doc.rust-lang.org/std/collections/hash_map/struct.DefaultHasher.html

"The internal algorithm is not specified, and so it and its hashes should not be relied upon over releases."

Does this mean this could change depending on the rust version? Or maybe some other differences, like platform?